viewing_party/party.py

```diff
@@ -1,23 +1,138 @@
+from statistics import mean, mode
+# -----------------------------------------
 # ------------- WAVE 1 --------------------
-
+# -----------------------------------------
+# If valid, creates a movie with title, genre, and rating as values of a dict
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        return None
+    else:
+        movie = {"title" : title, "genre" : genre, "rating" : rating}
+    return movie
+
+# Adds movie dictionary to watched
+def add_to_watched(user_data, movie):
+    user_data = {}
+    user_data["watched"] = [movie]
+    return user_data
+
+# Adds movie dictionary to watchlist
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+# If the title is in a movie in the user's watchlist: remove that movie from the watchlist, add that movie to watched
+def watch_movie(user_data, title):
+    watchlist = user_data["watchlist"]
+    for film in range(len(watchlist)):
+        if watchlist[film]["title"] == title:
+            user_data["watched"].append(watchlist[film])
+            del watchlist[film]
+    return user_data

```

Comment:
👍 

While ranged loops add more syntax to read than a `for in` loop, utilizing the index here to use `del` is a smart move. In general, we should be wary of iteration over a collection while modifying it, but del will ensure that indices are also deleted along with the object. This prevents potential Indexerrors as the loop continues.